src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectivenessHelicopter.cpp

```diff
@@ -174,6 +196,21 @@ void ActuatorEffectivenessHelicopter::updateSetpoint(const matrix::Vector<float,
 	}
 }
 
+float ActuatorEffectivenessHelicopter::getLinearServoOutput(float input) const
+{
+	input = math::constrain(input, -1.0f, 1.0f);
+
+	//servo output is calculated by normalizing input to arm rotation of CA_MAX_SVO_THROW degrees as full input for a linear throw
+	float svo_height = _geometry.max_servo_height * input;
+
+	if (!PX4_ISFINITE(svo_height)) {
+		svo_height = 0.0f;
+	}
```

Comment:
Why is there a specific check for NAN exactly here?
`_geometry.max_servo_height` is not randomly NAN because it's a parameter and there's no division by zero. Also it's static and wouldn't get NAN mid air. So I think you cover the case where `input` is NAN which I didn't go through in detail when it is but could to produce a disarmed value to the servo. In case it is nan you set the `servo_height` to 0 which results in a 0 output. Did you consider checking for NAN before processing anything and just forwarding NAN through instead or explicitly handle a NAN output setting it to zero? It's correct, I'm just curious if we can make the intention more obvious.

Reply:
We figured out in direct contact that this check is not needed because the value is never NAN and if it were there would be other issues with the swash plate even without the linearization and it'd need to be caught outside. So it was removed in c93f2a40d80655408db9f722f85d8eb0a3bbc5bc